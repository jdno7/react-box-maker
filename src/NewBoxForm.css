.NewBoxForm {
    margin: auto;
    width: 800px;
    /* text-align: center; */
}