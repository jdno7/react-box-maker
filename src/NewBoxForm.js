import './NewBoxForm.css'
import {useState} from 'react'

const NewBoxForm = ({addNewBox}) => {

    const [formData, setFormData] = useState({color:"", width:"", height:""})
    // const [form, setForm] = useState({color:"", width:"", height:""})
    // const [color, setColor] = useState()
    // const [width, setWidth] = useState()
    // const [height, setHeight] = useState()

    const handleChange = (e) => {
        const {name,value} = e.target
        setFormData(data => ({
            ...data,
            [name]: value
        })
        )}

    // const handleColorChange = (e) => {
    //     setColor(e.target.value)
    // }
    // const handleWidthChange = (e) => {
    //     setWidth(e.target.value)
    // }
    // const handleHeightChange = (e) => {
    //     setHeight(e.target.value)
    // }

    const handleSubmit = (e) => {
        e.preventDefault();
        // alert(`New Box Created! color:${formData.color} width:${formData.width} height:${formData.height}`)
        addNewBox(formData)
        setFormData({color:"", width:"", height:""})

    }

    return (
        <div className='NewBoxForm'>
            <h2>Add a New Box</h2>
            <form onSubmit={handleSubmit}>
                <div>
                    <label htmlFor='color'>Box Color:</label> <br/>
                        <input type="text" id="color" name='color' value={formData.color} onChange={handleChange}/>
                </div>
                <div>
                    <label htmlFor='width'>Box Width:</label> <br/>
                        <input type="text" id="width" name='width' value={formData.width} onChange={handleChange}/>
                </div>
                <div>
                <label htmlFor='height'>Box Height:</label> <br/>
                    <input type="text" id="height" name='height' value={formData.height} onChange={handleChange}/>
                </div>
                <button>Add Box!</button>
            </form>
        </div>
    )
}

export default NewBoxForm;