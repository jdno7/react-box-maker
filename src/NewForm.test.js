import { getByTestId, render, screen } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';
import {fireEvent} from '@testing-library/react'

test('renders BoxList Component', () => {
  render(<NewBoxForm />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
});

test('Snapshot Test', () => {
    const {asFragment} = render(<NewBoxForm/>)
    expect(asFragment()).toMatchSnapshot()
})