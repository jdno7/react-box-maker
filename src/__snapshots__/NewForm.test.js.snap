// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="NewBoxForm"
  >
    <h2>
      Add a New Box
    </h2>
    <form>
      <div>
        <label
          for="color"
        >
          Box Color:
        </label>
         
        <br />
        <input
          id="color"
          name="color"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="width"
        >
          Box Width:
        </label>
         
        <br />
        <input
          id="width"
          name="width"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="height"
        >
          Box Height:
        </label>
         
        <br />
        <input
          id="height"
          name="height"
          type="text"
          value=""
        />
      </div>
      <button>
        Add Box!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
